### ISSUE TITLE
Configuration: read from .babelrc (strict mode, etc)

### ISSUE THREAD

**Issue by chriswong** [babel/babel-eslint#171]
## with

```javascript
function foo() {
    "use strict";

    with (arguments) {
        length;
    }
    return arguments.caller;
}
foo();
```
> test.js
>  4:4  error  'with' in strict mode
>
> ✖ 1 problem (1 error, 0 warnings)

## arguments.caller

```javascript
function foo() {
    "use strict";

    // with (arguments) {
    //     length;
    // }
    return arguments.caller;
}
foo();
```

> format/test.js
>   2:4   error  Unnecessary 'use strict'  no-extra-strict
>   7:11  error  Avoid arguments.caller    no-caller
> 
> ✖ 2 problems (2 errors, 0 warnings)

**Comment by hzoo** [comment]
Can you explain more about what the issue is? I'm assuming you're saying that when you use `with` it causes a different error than than the ESLint rule errors? The first error is from babel since it's strict mode by default.

**Comment by otakustay** [comment]
@hzoo `with` and `arguments.caller` are both strict mode restrictions, however if my code contains a `with` statement, babel-eslint does not generate the error message about `arguments.caller`, we expect it generate 2 errors (`with` and `arguments.caller`) for the first code example

**Comment by hzoo** [comment]
I think I mentioned this above but the first error is from the babel parser (since ESLint doesn't prevent the use of with unless you use `no-with`, and the error message is different from ESLint). In the second case the parser doesn't error so the ESLint errors appear. Not sure what we can do here

It's the same on the [repl](http://babeljs.io/repl/#?experimental=true&evaluate=false&loose=false&spec=false&code=function%20foo()%20%7B%0A%20%20%20%20%22use%20strict%22%3B%0A%0A%20%20%20%20with%20(arguments)%20%7B%0A%20%20%20%20%20%20%20%20length%3B%0A%20%20%20%20%7D%0A%20%20%20%20return%20arguments.caller%3B%0A%7D%0Afoo()%3B).

**Comment by otakustay** [comment]
I tried to understand this case, does it means babel parser treats use of `with` as an error so it stops further parsing of the source code, which as a result does not generate the error of `arguments.caller`?

**Comment by chriswong** [comment]
Yes, i think it treats use of `with` and `delete` as syntax error.

**Comment by otakustay** [comment]
I think I can describe the issue more clearly now

Our problem is that for the first code example, even we remove the `"use strict"` statement, babel-eslint still generate 1 error for the `with` statement

The actual cause is babel parser will automatically treat code as strict mode code, we can add `blacklist: ['strict']` configuration to avoid it, however we don't find any way to add this config in eslint or babel-eslint, so is there any API for us to configure babel-parser's behavior in babel-eslint?

Thanks

**Comment by hzoo** [comment]
Right, yeah there currently isn't any way to configure how babel-eslint works.

**Comment by hzoo** [comment]
The way babel-eslint is hooked in is through the `parser` config option http://eslint.org/docs/user-guide/configuring#specifying-parser. So unless we do a better integration I don't think we can do anything? @sebmck

**Comment by otakustay** [comment]
How about babel-parser reads config from `.babelrc`, we can specify configurations there

**Comment by danez** [comment]
in the next version babel-eslint will respect when you set the `sourceType` in eslint to `script`, which disables global strictmode. With `sourceType` set to `module`, strict mode is always on by spec.
```
{
    "parserOptions": {
        "sourceType": "module"
    }
}
```
